src/main/java/org/opentripplanner/graph_builder/module/ned/ElevationModule.java

```diff
@@ -220,10 +247,68 @@ public void buildGraph(Graph graph, GraphBuilderModuleSummary graphBuilderModule
-        assignMissingElevations(graph, edgesWithElevation);
+        assignMissingElevations(graph, edgesWithCalculatedElevations);
         log.info(missingElevationsTask.finish());
     }
 
+    /**
+     * A runnable that contains the relevant info for executing a process edge operation in a particular thread.
+     */
+    private class ProcessEdgeTask implements Runnable {
+        private final StreetWithElevationEdge swee;
+        private final ConcurrentHashMap<Long, Coverage> coveragesForThread;
+
+        public ProcessEdgeTask(StreetWithElevationEdge swee, ConcurrentHashMap<Long, Coverage> coveragesForThread) {
+            this.swee = swee;
+            this.coveragesForThread = coveragesForThread;
+        }
+
+        @Override public void run() {
+            processEdge(swee, getThreadSpecificCoverageInstance());
+        }
+
+        /**
+         * Get the thread-specific Coverage instance to avoid multiple threads waiting for a lock to be released on a
+         * Interpolator2D instance.
+         */
+        private Coverage getThreadSpecificCoverageInstance () {
+            long currentThreadId = Thread.currentThread().getId();
+            Coverage threadSpecificCoverage = coveragesForThread.get(currentThreadId);
+            if (threadSpecificCoverage == null) {
+                // Synchronize the creation of the Thread-specific Coverage instances to avoid potential deadlocks that
+                // could arise from downstream classes that have synchronized methods.
+                synchronized (coveragesForThread) {
+                    threadSpecificCoverage = getCoverage();
+                    // The Coverage instance relies on some synchronized static methods shared across all threads that
+                    // can cause deadlocks if not fully initialized. Therefore, make a single request for the first
+                    // point on the edge to initialize these other items.
+                    Coordinate firstEdgeCoord =  swee.getGeometry().getCoordinates()[0];
+                    double[] dummy = new double[1];
+                    threadSpecificCoverage.evaluate(
+                        new DirectPosition2D(GeometryUtils.WGS84_XY, firstEdgeCoord.x, firstEdgeCoord.y),
+                        dummy
+                    );
+                }
+                coveragesForThread.put(currentThreadId, threadSpecificCoverage);
+            }
+            return threadSpecificCoverage;
+        }
+    }
+
+    /**
+     * Get an uncached Coverage instance from the module's ElevationGridCoverageFactory.
+     */
+    private Coverage getCoverage() {
+        // If gridCov is a GridCoverage2D, apply a bilinear interpolator. Otherwise, just use the
+        // coverage as is (note: UnifiedGridCoverages created by NEDGridCoverageFactoryImpl handle
+        // interpolation internally)
+        Coverage gridCov = gridCoverageFactory instanceof NEDGridCoverageFactoryImpl
```

Comment:
This comment and how it applies to the method body is a bit confusing. I think it needs to be updated based on your refactor. Also, I don't love the competing ternary styles here.